pkg/occlient/occlient.go

```diff
@@ -1015,3 +973,189 @@ func (c *Client) ListRouteNames(labelSelector string) ([]string, error) {
 
 	return routeNames, nil
 }
+
+// CreatePVC creates a PVC resource in the cluster with the given name, size and
+// labels
+func (c *Client) CreatePVC(name string, size string, labels map[string]string) (*corev1.PersistentVolumeClaim, error) {
+	quantity, err := resource.ParseQuantity(size)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse size: %v", size)
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: labels,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: quantity,
+				},
+			},
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+		},
+	}
+
+	createdPvc, err := c.kubeClient.CoreV1().PersistentVolumeClaims(c.namespace).Create(pvc)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create PVC")
+	}
+	return createdPvc, nil
+}
+
+// DeletePVC deletes the given PVC by name
+func (c *Client) DeletePVC(name string) error {
+	return c.kubeClient.CoreV1().PersistentVolumeClaims(c.namespace).Delete(name, nil)
+}
+
+// generateVolumeNameFromDC generates a random volume name based on the name
+// of the given Deployment Config
+func generateVolumeNameFromDC(dc string) string {
+	return fmt.Sprintf("%v-%v-volume", dc, util.GenerateRandomString(nameLength))
+}
+
+// AddPVCToDeploymentConfig adds the given PVC to the given Deployment Config
+// at the given path
+func (c *Client) AddPVCToDeploymentConfig(dc *appsv1.DeploymentConfig, pvc string, path string) error {
+	volumeName := generateVolumeNameFromDC(dc.Name)
```

Comment:
Why does volume name have random value in it? Volume is inside DC so it can only collide with other volumes for that DC, but not with anything else.
Wouldn't it be better to name it after PVC?

Reply:
makes sense, changed this to derive from PVC